### ISSUE TITLE
Crash on _deserializeMessageJSON

### ISSUE THREAD

**Issue by peterpaulis** [marcuswestin/WebViewJavascriptBridge#159]
Through Crashlytics we encountered this problem, any idea?

<pre>
Thread : Fatal Exception: NSInvalidArgumentException
0  CoreFoundation                 6515134280 __exceptionPreprocess
1  libobjc.A.dylib                6862946176 objc_exception_throw
2  CoreFoundation                 6515134096 -[NSException initWithCoder:]
3  Foundation                     6530637148 +[NSJSONSerialization JSONObjectWithData:options:error:]
4  App - AdHoc               4295319320 -[WebViewJavascriptBridgeBase _deserializeMessageJSON:] (WebViewJavascriptBridgeBase.m:208)
5  App - AdHoc               4295315472 -[WebViewJavascriptBridgeBase flushMessageQueue:] (WebViewJavascriptBridgeBase.m:66)
6  App - AdHoc               4295577020 __48-[WKWebViewJavascriptBridge WKFlushMessageQueue]_block_invoke (WKWebViewJavascriptBridge.m:94)
7  WebKit                         6620893508 std::__1::__function::__func<-[WKWebView evaluateJavaScript:completionHandler:]::$_0, std::__1::allocator<-[WKWebView evaluateJavaScript:completionHandler:]::$_0>, void (API::SerializedScriptValue*, bool, WebKit::CallbackBase::Error)>::operator()(API::SerializedScriptValue*&&, bool&&, WebKit::CallbackBase::Error&&)
8  WebKit                         6620067680 WebKit::GenericCallback<API::SerializedScriptValue*, bool>::invalidate(WebKit::CallbackBase::Error)
9  WebKit                         6619089392 void WebKit::invalidateCallbackMap<WTF::RefPtr<WebKit::CallbackBase> >(WTF::HashMap<unsigned long long, WTF::RefPtr<WebKit::CallbackBase>, WTF::IntHash<unsigned long long>, WTF::HashTraits<unsigned long long>, WTF::HashTraits<WTF::RefPtr<WebKit::CallbackBase> > >&, WebKit::CallbackBase::Error)
10 WebKit                         6619979964 WebKit::WebPageProxy::resetState(WebKit::WebPageProxy::ResetStateReason)
11 WebKit                         6619993976 WebKit::WebPageProxy::resetStateAfterProcessExited()
12 WebKit                         6620025872 WebKit::WebPageProxy::processDidCrash()
13 WebKit                         6620349936 WebKit::WebProcessProxy::didClose(IPC::Connection&)
14 JavaScriptCore                 6544093888 WTF::RunLoop::performWork()
15 JavaScriptCore                 6544095084 WTF::RunLoop::performWork(void*)
16 CoreFoundation                 6514836804 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__
17 CoreFoundation                 6514835416 __CFRunLoopDoSources0
18 CoreFoundation                 6514826456 __CFRunLoopRun
19 CoreFoundation                 6513970336 CFRunLoopRunSpecific
20 GraphicsServices               6700859528 GSEventRunModal
21 UIKit                          6605295612 UIApplicationMain
22 App - AdHoc               4295638056 main (main.m:14)
23 libdyld.dylib                  6871656632 start
</pre>

**Comment by stasnikiforov** [comment]
I found that through Crashlytics as well. So I did this:
```objective-c
    NSData *data = [messageJSON dataUsingEncoding:NSUTF8StringEncoding];
    if (data) {
        return [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingAllowFragments error:nil];
    } else {
#ifdef USE_CRASHLYTICS
        [Answers logCustomEventWithName:@"empty-js" customAttributes:@{@"message": (messageJSON ?: @"nil message")}];
#endif
    }
    return nil;
```

And every time I see the event, the message param is "nil message". Not sure why messageJSON would be nil.

**Comment by marcuswestin** [comment]
Hi! We have automated tests now - I would love to see a repro of this.

You can run the tests with `make test`, or by opening `Tests/WebViewJavascriptBridge.xcodeproj` and then Product > Test.

Try creating a repro that fails in there! As soon as you do I'm confident that we can nail this guy down.

Cheers!

**Comment by zhipcui** [comment]
maybe related to the #183 issues

**Comment by marcuswestin** [comment]
Fixed in v5.0.2
